kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/SaslAuthenticator.java

```diff
@@ -328,8 +328,11 @@ private static ByteBuf sizePrefixed(ByteBuffer buffer) {
         ByteBuffer sizeBuffer = ByteBuffer.allocate(4);
         sizeBuffer.putInt(0, buffer.remaining());
         ByteBuf byteBuf = Unpooled.buffer(sizeBuffer.capacity() + buffer.remaining());
+        // why we reset writer index? see https://github.com/streamnative/kop/issues/696
+        byteBuf.markWriterIndex();
         byteBuf.writeBytes(sizeBuffer);
         byteBuf.writeBytes(buffer);
+        byteBuf.resetWriterIndex();
```

Comment:
We can make `sizePrefixed` public and add a related test. For example, pass a `ByteBuffer` with

```java
        final ByteBuffer byteBuffer = ByteBuffer.allocate(8);
        for (int i = 0; i < byteBuffer.capacity(); i++) {
            byteBuffer.put((byte) i);
        }
        byteBuffer.position(4);
```

Then you can verify the reader index, writer index, capacity.

Reply:
@BewareMyPower Thanks for your review. Good suggestion. I have added a test. PTAL :)